Build failing with LDFLAGS --as-needed

When building with `LDFLAGS=-Wl,--as-needed`, it fails at linking stage:

```
libtool: link: g++ -DSAPI_CLIENT -I. -g -O2 -Wl,--as-needed -o src/tpm2_listpcrs src/tpm2_listpcrs.o  -ltss2 -ltctisocket src/libcommon.a
src/libcommon.a(common.o): In function `InitTctiResMgrContext':
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:224: undefined reference to `InitSocketTcti'
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:232: undefined reference to `InitSocketTcti'
src/libcommon.a(common.o): In function `Cleanup':
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:250: undefined reference to `PlatformCommand'
src/libcommon.a(common.o): In function `TeardownTctiResMgrContext':
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:243: undefined reference to `TeardownSocketTcti'
src/libcommon.a(common.o): In function `prepareTest':
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:287: undefined reference to `PlatformCommand'
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:288: undefined reference to `PlatformCommand'
src/libcommon.a(common.o): In function `TeardownTctiResMgrContext':
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:243: undefined reference to `TeardownSocketTcti'
/tmp/yaourt-tmp-hexchain/aur-tpm2.0-tools-git/src/tpm2.0-tools/src/common.c:243: undefined reference to `TeardownSocketTcti'
collect2: error: ld returned 1 exit status
Makefile:813: recipe for target 'src/tpm2_listpcrs' failed
make: *** [src/tpm2_listpcrs] Error 1
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -36,7 +36,7 @@ INCLUDE_DIRS = -I$(srcdir)
 
 AM_CFLAGS   = -DSAPI_CLIENT $(INCLUDE_DIRS)
 AM_CXXFLAGS = -DSAPI_CLIENT $(INCLUDE_DIRS)
-LDADD = -ltss2 -ltctisocket src/libcommon.a
+LDADD = src/libcommon.a -ltss2 -ltctisocket
 
 noinst_LIBRARIES = src/libcommon.a
 sbin_PROGRAMS = src/tpm2_listpcrs \

